### PR TITLE
Add authentication options

### PR DESCRIPTION
**Pull request by duboisph**
Adds options for enabling authentication

### CODE REVIEW

File: redis/variables.tf [hunk 1]
```diff
@@ -71,3 +71,13 @@ variable "snapshot_arns" {
...
 }
+
+variable "transit_encryption_enabled" {
+  description = "(Optional) Whether to enable encryption in transit"
+  default     = false
+}
+
+variable "auth_token" {
+  description = "(Optional) The password used to access a password protected server. Can be specified only if `transit_encryption_enabled = true`"
+  default     = ""
```
Comment (MattiasGees) [inline redis/variables.tf:87]: This is tested when it is empty?

Comment (duboisph) [reply redis/variables.tf:87]: argh, bit too fast

File: README.md [hunk 2]
```diff
@@ -21,14 +21,16 @@ This creates a redis cluster with some default values and creates a security gro
  * [`snapshot_window`]: String: The daily time range (in UTC) during which ElastiCache will begin taking a daily snapshot of your cache cluster. The minimum maintenance window is a 60 minute period. Example: 05:00-09:00
  * [`snapshot_retention_limit`]: String: The number of days for which ElastiCache will retain automatic cache cluster snapshots before deleting them. For example, if you set SnapshotRetentionLimit to 5, then a snapshot that was taken today will be retained for 5 days before being deleted. If the value of SnapshotRetentionLimit is set to zero (0), backups are turned off. Please note that setting a snapshot_retention_limit is not supported on cache.t1.micro or cache.t2.* cache nodes
  * [`snapshot_arns`]: List(Optional): A single-element string list containing an Amazon Resource Name (ARN) of a Redis RDB snapshot file stored in Amazon S3. Example: arn:aws:s3:::my_bucket/snapshot1.rdb
+ * [`transit_encryption_enabled`]: Bool(Optional) Whether to enable encryption in transit"
```
Comment (simonrondelez) [inline README.md]: you forgot to add that it defaults to false

**Review by simonrondelez — CHANGES_REQUESTED** [review]

**Review by MattiasGees — APPROVED** [review]

**Review by simonrondelez — APPROVED** [review]

**Review by MattiasGees — APPROVED** [review]
Look ok, don't forget to major bump it ;)

**Pull request merged**